bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/handler/IntesisHomeHandler.java

```diff
@@ -0,0 +1,566 @@
+import org.openhab.binding.intesis.internal.IntesisDynamicStateDescriptionProvider;
+import org.openhab.binding.intesis.internal.IntesisHomeModeEnum;
+import org.openhab.binding.intesis.internal.api.IntesisHomeHttpApi;
+import org.openhab.binding.intesis.internal.gson.IntesisHomeJSonDTO.data;
+import org.openhab.binding.intesis.internal.gson.IntesisHomeJSonDTO.datapoints;
+import org.openhab.binding.intesis.internal.gson.IntesisHomeJSonDTO.descr;
+import org.openhab.binding.intesis.internal.gson.IntesisHomeJSonDTO.dp;
+import org.openhab.binding.intesis.internal.gson.IntesisHomeJSonDTO.dpval;
+import org.openhab.binding.intesis.internal.gson.IntesisHomeJSonDTO.id;
+import org.openhab.binding.intesis.internal.gson.IntesisHomeJSonDTO.info;
+import org.openhab.binding.intesis.internal.gson.IntesisHomeJSonDTO.response;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link IntesisHomeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Hans-Jörg Merk - Initial contribution
+ */
+@NonNullByDefault
+public class IntesisHomeHandler extends BaseThingHandler implements DynamicStateDescriptionProvider {
+
+    private final Logger logger = LoggerFactory.getLogger(IntesisHomeHandler.class);
+    private final IntesisHomeHttpApi api;
+    private IntesisConfiguration config = new IntesisConfiguration();
+
+    private final Map<ChannelUID, @Nullable StateDescription> descriptions = new ConcurrentHashMap<>();
+    private final Map<String, String> properties = new HashMap<>();
+
+    private IntesisDynamicStateDescriptionProvider intesisStateDescriptionProvider;
+
+    private @Nullable ScheduledFuture<?> refreshJob;
+
+    private String ipAddress = "";
+    private String password = "";
+    private String sessionId = "";
+
+    final Gson gson = new Gson();
+
+    public IntesisHomeHandler(final Thing thing, final HttpClient httpClient,
+            IntesisDynamicStateDescriptionProvider intesisStateDescriptionProvider) {
+        super(thing);
+        this.api = new IntesisHomeHttpApi(config, httpClient);
+        this.intesisStateDescriptionProvider = intesisStateDescriptionProvider;
+    }
+
+    private static String beautify(final String camelCaseWording) {
+        final StringBuilder b = new StringBuilder();
+        for (final String s : StringUtils.splitByCharacterTypeCamelCase(camelCaseWording)) {
+            b.append(" ");
+            b.append(s);
+        }
+        final StringBuilder bs = new StringBuilder();
+        for (final String t : StringUtils.splitByWholeSeparator(b.toString(), " _")) {
+            bs.append(" ");
+            bs.append(t);
+        }
+
+        return WordUtils.capitalizeFully(bs.toString()).trim();
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Start initializing!");
+        updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.CONFIGURATION_PENDING);
```

Comment:
```suggestion
        updateStatus(ThingStatus.UNKNOWN);
```
See https://www.openhab.org/docs/concepts/things.html#status-details

Reply:
Removed